blog/zh/blog/2022/12/27/apisix-ingress-with-Flagger.md

```diff
@@ -0,0 +1,403 @@
+    Reason:                Succeeded
+    Status:                True
+    Type:                  Promoted
+  Failed Checks:           1
+  Iterations:              0
+  Phase:                   Succeeded
+
+Events:
+  Type     Reason  Age                    From     Message
+  ----     ------  ----                   ----     -------
+  Warning  Synced  2m59s                  flagger  podinfo-primary.test not ready: waiting for rollout to finish: observed deployment generation less than desired generation
+  Warning  Synced  2m50s                  flagger  podinfo-primary.test not ready: waiting for rollout to finish: 0 of 1 (readyThreshold 100%) updated replicas are available
+  Normal   Synced  2m40s (x3 over 2m59s)  flagger  all the metrics providers are available!
+  Normal   Synced  2m39s                  flagger  Initialization done! podinfo.test
+  Normal   Synced  2m20s                  flagger  New revision detected! Scaling up podinfo.test
+  Warning  Synced  2m (x2 over 2m10s)     flagger  canary deployment podinfo.test not ready: waiting for rollout to finish: 0 of 1 (readyThreshold 100%) updated replicas are available
+  Normal   Synced  110s                   flagger  Starting canary analysis for podinfo.test
+  Normal   Synced  109s                   flagger  Advance podinfo.test canary weight 10
+  Warning  Synced  100s                   flagger  Halt advancement no values found for apisix metric request-success-rate probably podinfo.test is not receiving traffic: running query failed: no values found
+  Normal   Synced  90s                    flagger  Advance podinfo.test canary weight 20
+  Normal   Synced  80s                    flagger  Advance podinfo.test canary weight 30
+  Normal   Synced  69s                    flagger  Advance podinfo.test canary weight 40
+  Normal   Synced  59s                    flagger  Advance podinfo.test canary weight 50
+  Warning  Synced  30s (x2 over 40s)      flagger  podinfo-primary.test not ready: waiting for rollout to finish: 1 old replicas are pending termination
+  Normal   Synced  9s (x3 over 50s)       flagger  (combined from similar events): Promotion completed! Scaling down podinfo.test
+```
+
+在新版本金丝雀发布的过程中，你可以通过域名 app.example.com 访问应用（示例中的`app.example.com`可以替换成你的实际域名），这里将会出现不同版本的响应切换。
+
+![filename (3).png](https://static.apiseven.com/2022/12/26/63a9a47b281b0.png)
+
+通过查看由 Flagger 自动创建出来的 Apache APISIX 的`ApisixRoute` 资源 `podinfo-podinfo-canary`，会发现 service `podinfo-primary` 和 service `podinfo-canary` 的权重跟随着发布过程一起变化。
+
+```yaml
+spec:
+  http:
+    - backends:
+        - serviceName: podinfo-primary
+          servicePort: 80
+          # Flagger自动调整
+          weight: 80
+        - serviceName: podinfo-canary
+          servicePort: 80
+          # Flagger自动调整
+          weight: 20
+```
+
+当最终发布完成后，将会看到稳定的最新版本。
+
+![filename (4).png](https://static.apiseven.com/2022/12/26/63a9a479d5bbe.png)
+
+>**注意：**如若在金丝雀发布期间再次变更了 deployment，Flagger 将会重新进行金丝雀分析。你可以通过以下命令来观察到所有的金丝雀发布细节。
```

Comment:
```suggestion
:::note 注意

如若在金丝雀发布期间再次变更了 deployment，Flagger 将会重新进行金丝雀分析。你可以通过以下命令来观察到所有的金丝雀发布细节。

:::
```

Reply:
you can refer this
<img width="1291" alt="image" src="https://user-images.githubusercontent.com/39793568/209927027-28addb8e-ad06-42e5-b71c-e180b558cc77.png">